src/univariate/continuous/maximum.jl

```diff
@@ -0,0 +1,25 @@
+"""
+The maximum of n iid random variables with continuous univariate distribution
+"""
+struct Maximum{D<:ContinuousUnivariateDistribution} <: ContinuousUnivariateDistribution
+    dist::D
+    n::Int
+    Maximum{D}(dist, n) where {D<:ContinuousUnivariateDistribution} = new{D}(dist, n)
+end
+
+function Maximum(dist::D, n::Integer; check_args::Bool=true) where {D <: ContinuousUnivariateDistribution}
+    @check_args Maximum (n, n >= one(n))
+    return Maximum{D}(dist, n)
+end
+
+rand(rng::AbstractRNG, d::Maximum{D}) where {D} = maximum([rand(rng, d.dist) for _ in 1:d.n])
```

Comment:
Even better, since `rand(Beta(n, 1)` is equivalent to `rand()^(1//n)`
```suggestion
rand(rng::AbstractRNG, d::Maximum) = maximum(rand(rng, d.dist) for _ in 1:d.n)
function rand(rng::AbstractRNG, d::Maximum{<:ContinuousUnivariateDistribution})
    return quantile(d.dist, rand(rng)^(1//d.n))
end
```